book/09-git-and-other-scms/sections/import-svn.asc

```diff
@@ -104,7 +104,7 @@ $ for b in $(git for-each-ref --format='%(refname:short)' refs/remotes); do git
 ----
 
 It may happen that you'll see some extra branches which are suffixed by `@xxx` (where xxx is a number), while in Subversion you only see one branch.
-This is actually a Subversion feature called ``peg-revisions'', which is something that Git simply has no syntactical counterpart for.
+This is actually a Subversion feature called `peg-revisions`, which is something that Git simply has no syntactical counterpart for.
```

Comment:
🤔 We only set a name in monospace if it's a literal term used in a code editor or a terminal. I don't know SVN that well, but my google-fu doesn't turn up any use of a `peg-revision` literal. If I'm right, we should change this to ` ``peg revisions'' `.

Reply:
The problem with the old revision was that all text starting with ``peg-revisions'' was displayed as an inline code-block because the code block marks were not ended properly

Reply:
That may be true with the GitHub asciidoc renderer, but that's not our main front-end. The HTML version that renders on git-scm.com looks like this:

![image](https://user-images.githubusercontent.com/39902/67133110-3a8f5e80-f1c0-11e9-90e0-8c0b90b2794d.png)

And the PDF version looks like this:

![image](https://user-images.githubusercontent.com/39902/67133123-5561d300-f1c0-11e9-8965-a572438460f5.png)

Both of which are exactly what we want.

I do wonder if the GitHub renderer would agree with the other outputs if we used the `` "`newer syntax`" `` mentioned in [the Asciidoctor documentation](https://asciidoctor.org/docs/asciidoc-syntax-quick-reference/).

Reply:
I already mentioned a similar issue in [this comment](https://github.com/progit/progit2/pull/1301#issuecomment-529828540).

Reply:
So which of the mentioned alternatives would you suggest, @max123kl

Reply:
Unfortunately, this depends on the editor and/or the operating system used.
For macOS and Windows you can use the following table.
![quotation_marks](https://user-images.githubusercontent.com/853497/67140886-1a62ac80-f25f-11e9-8e77-364f8077761e.png)
The complete list can be found [here](https://www.key-shortcut.com/en/all-html-entities/all-entities) (look for quotation and/or punctuation)
On Linux machines you can program a keyboard shortcut or copy+paste the Unicode special characters from a second file.

Reply:
On Linux you can hold right Ctrl+right Shift+u and then type x201C
Its a bit painful but works

Reply:
> I already mentioned a similar issue in [this comment](https://github.com/progit/progit2/pull/1301#issuecomment-529828540).

And in that thread I mentioned that I don't want to adopt "frozen" curly quotes in the source code for this book:

> In the end, it's much easier for new contributors to type the standard ASCII characters, regardless of their keyboard layout, OS, or editor, so we should stick to either the old (` ``quote'' `) or new (`` "`quote`" ``) Asciidoc notation.

To that end, please replace this change with ` ``peg revisions'' `. That's consistent with our current style, and renders properly everywhere but on GitHub, which is not a recommended way to read the book.

Reply:
I do respect your decision in this repo.

But I don't think I understand what you mean by "frozen". Is anything be blocked or locked?

The key code I do propose in my last post will produce the typographically correct quotes, not the straight ones. That's what you want, isn't it?

Because the punctuation (quotation marks) in German, as well as in other foreign languages, is visually different from here, you have to find an alternative, working solution.
Fortunately, the UTF-8 code offers all required characters. So you can, without conversion via the text editor, get the desired characters by key code.

In the German Repo I implemented my suggestion and didn't detect any error in any of the important output formats. On the other hand, the GitHub renderer also displays the quotation marks correctly.

**DE_quotation_HTML**
![DE_quotation_HTML](https://user-images.githubusercontent.com/853497/67156395-43964200-f31e-11e9-8d5a-58d3990a27e0.png)

**DE_quotation_PDF**
![DE_quotation_PDF](https://user-images.githubusercontent.com/853497/67156423-8d7f2800-f31e-11e9-967a-e7705b6a3db2.png)

**DE_quotation_ePub**
![DE_quotation_ePub](https://user-images.githubusercontent.com/853497/67156409-5f99e380-f31e-11e9-8852-758fb35ca6e1.png)

**DE_quotation_mobi**
![DE_quotation_mobi](https://user-images.githubusercontent.com/853497/67156418-76d8d100-f31e-11e9-9f19-070e304b3c31.png)

**DE_quotation_GitHub**
![DE_quotation_GitHub](https://user-images.githubusercontent.com/853497/67156321-33ca2e00-f31d-11e9-960f-3a859c5639df.png)

Reply:
> I don't think I understand what you mean by "frozen".

I just mean that the "curliness" of the quotation mark is embedded in the choice of utf-8 character, where the `` "` `` notation makes a source-code token that _renders_ to a curly quote. 

We did the opposite of what you're doing in your repo with en- and em-dashes a while back; the literal "–" character was un-rendered to `--` in the source code so it would be easier to type, and harder to mistake for a hyphen.

I like having source code that can render to a configurable target character. If we decide to convert all dashes to hyphens in the future, we can do that in the renderer. If we decide we want all quotation marks to be straight, we can do that without having to `sed` through the entire source code.

Oh, and as it turns out, the GitHub renderer handles the `` "`new style`" `` notation [properly](https://gist.github.com/ben/7de7719bfc14967f2094004c8974a583).